base/public.jl

```diff
@@ -62,6 +62,8 @@ public
     isexported,
     ispublic,
     remove_linenums!,
+    method_argnames,
```

Comment:
Fair about the `#self` piece, but it would be really nice to be able to get the arg names. 

Do you think cut it completely or define a new function that could provide just the arg names (But then there's the issue of functions defined for callable structs where you actually do want the function "self"...)

Reply:
Perhaps replace `Symbol("#self#")` and `Symbol("#unused#")` with `nothing`?

Reply:
I'll give it a go!